package/MDAnalysis/analysis/bat.py

```diff
@@ -0,0 +1,503 @@
+for a given set of atoms or residues. This coordinate system is designed
+to be complete, non-redundant, and minimize correlations between degrees
+of freedom. Complete and non-redundant means that for N atoms there will
+be 3N Cartesian coordinates and 3N BAT coordinates. Correlations are
+minimized by using improper torsions described in [Hikiri2016]_.
+
+More specifically, bond refers to the bond length, or distance between
+a pair of bonded atoms. Angle refers to the bond angle, the angle between
+a pair of bonds to a central atom. Torsion refers to the torsion angle.
+For a set of four atoms a, b, c, and d, a torsion requires bonds between
+a and b, b and c, and c and d. The torsion is the angle between a plane
+containing atoms a, b, and c and another plane containing b, c, and d.
+For a set of torsions that share atoms b and c, one torsion is defined as
+the primary torsion. The others are defined as improper torsions, differences
+between the raw torsion angle and the primary torsion. This definition reduces
+the correlation between the torsion angles.
+
+Each molecule also has six external coordinates that define its translation and
+rotation in space. The three Cartesian coordinates of the first atom are the
+molecule's translational degrees of freedom. Rotational degrees of freedom are
+specified by the axis-angle convention. The rotation axis is a normalized vector
+pointing from the first to second atom. It is described by the polar angle,
+:math:`phi`, and azimuthal angle, :math:`theta`. :math:`omega` is a third angle
+that describes the rotation of the third atom about the axis.
+
+This module was adapted from AlGDock [Minh2020]_.
+
+
+See Also
+--------
+:class:`~MDAnalysis.analysis.dihedrals.Dihedral`
+   class to calculate dihedral angles for a given set of atoms or residues
+:func:`MDAnalysis.lib.distances.calc_dihedrals()`
+   function to calculate dihedral angles from atom positions
+
+
+Example applications
+--------------------
+
+The :class:`~MDAnalysis.analysis.bat.BAT` class defines bond-angle-torsion
+coordinates based on the topology of an atom group and interconverts between
+Cartesian and BAT coordinate systems. For example, we can determine internal
+coordinates for residues 5-10 of adenylate kinase (AdK). The trajectory is
+included within the test data files::
+
+   import MDAnalysis as mda
+   from MDAnalysisTests.datafiles import PSF, DCD
+   import numpy as np
+
+   u = mda.Universe(PSF, DCD)
+
+   # selection of atomgroups
+   selected_residues = u.select_atoms("resid 5-10")
+
+   from MDAnalysis.analysis.bat import BAT
+   R = BAT(selected_residues)
+
+   # Calculate BAT coordinates for a trajectory
+   R.run()
+
+   # Reconstruct Cartesian coordinates from BAT coordinates
+   # of the first frame
+   bat = R.bat[0]
+   XYZ = R.Cartesian(bat)
+
+   # The difference between the original and reconstructed coordinates
+   # should be zero.
+   print(np.sum(np.abs(XYZ - selected_residues.positions)>1E-6))
```

Comment:
Also show `save`/`load`.

Reply:
Should I keep the names load_bat and save_bat?

Reply:
No, call them `load()` and `save()`.